addons/captives/functions/fnc_doBlindfoldCaptive.sqf

```diff
@@ -0,0 +1,43 @@
+#include "script_component.hpp"
+/*
+ * Author: mrschick
+ * Puts a blindfold on a captive unit if the player has a blindfold in their inventory
+ *
+ * Arguments:
+ * 0: _unit-Player <OBJECT>
+ * 1: target <OBJECT>
+ *
+ * Return Value:
+ * The return value <BOOL>
+ *
+ * Example:
+ * [player, bob, true] call ACE_captives_fnc_doBlindfoldCaptive;
+ *
+ * Public: No
+ */
+
+params ["_unit", "_target", "_state"];
+
+// check if _unit has a blindfold in its inventory, abort otherwise.
+private _carriedBlindfoldIdx = GVAR(blindfolds) findAny (_unit call EFUNC(common,uniqueItems));
+if (_carriedBlindfoldIdx == -1) exitWith { ERROR("no blindfold"); };
+
+_unit removeItem (GVAR(blindfolds) select _carriedBlindfoldIdx);
+
+// Remove _target goggles if it is wearing any and move them to _unit or _target inventory (if they can hold them)
+private _previousGoggles = goggles _target;
+if (_previousGoggles != "") then {
+    if ([_unit, _previousGoggles] call CBA_fnc_canAddItem) exitWith {
+        removeGoggles _target;
+        _unit addItem _previousGoggles;
+    };
+    if ([_target, _previousGoggles] call CBA_fnc_canAddItem) exitWith {
+        removeGoggles _target;
+        _target addItem _previousGoggles;
+    };
+    // if the target goggles can fit in neither _unit nor _target inventory, drop them on the ground
+    [_target, _previousGoggles] call CBA_fnc_dropItem;
+};
+
+_target addGoggles (GVAR(blindfolds) select _carriedBlindfoldIdx);
+
```

Comment:
```suggestion
#include "script_component.hpp"
/*
 * Author: mrschick, johnb43
 * Puts a blindfold on a captive unit if the player has a blindfold in their inventory.
 *
 * Arguments:
 * 0: Unit <OBJECT>
 * 1: Target <OBJECT>
 * 2: Put on (true) or take off (false) <BOOL>
 *
 * Return Value:
 * None
 *
 * Example:
 * [player, cursorTarget, true] call ace_captives_fnc_doBlindfoldCaptive
 *
 * Public: No
 */

params ["_unit", "_target", "_state"];

// Blindfold target
if (_state) then {
    // Check if _unit has a blindfold in its inventory, abort otherwise.
    private _carriedBlindfoldIdx = GVAR(blindfolds) findAny (_unit call EFUNC(common,uniqueItems));
    if (_carriedBlindfoldIdx == -1) exitWith {};

    private _blindfold = GVAR(blindfolds) select _carriedBlindfoldIdx;

    _unit removeItem _blindfold;

    // Remove target's goggles if it is wearing any and move them to unit's or target's inventory (if they can hold them)
    private _previousGoggles = goggles _target;

    if (_previousGoggles != "") then {
        if ([_unit, _previousGoggles] call CBA_fnc_canAddItem) exitWith {
            removeGoggles _target;
            _unit addItem _previousGoggles;
        };

        if ([_target, _previousGoggles] call CBA_fnc_canAddItem) exitWith {
            removeGoggles _target;
            _target addItem _previousGoggles;
        };

        // If the target's goggles can fit in neither unit's nor target's inventory, drop them on the ground
        private _weaponHolder = nearestObject [_target, "WeaponHolder"];

        if (isNull _weaponHolder || {_target distance _weaponHolder > 2}) then {
            _weaponHolder = createVehicle ["GroundWeaponHolder", [0, 0, 0], [], 0, "NONE"];
            _weaponHolder setPosASL getPosASL _target;
        };

        _weaponHolder addItemCargoGlobal [_previousGoggles, 1];
    };

    _target addGoggles _blindfold;
} else {
    // Remove blindfold from target
    private _previousGoggles = goggles _target;

    // If already not wearing blindfold, ignore
    if !(_previousGoggles in GVAR(blindfolds)) exitWith {};

    if ([_unit, _previousGoggles] call CBA_fnc_canAddItem) exitWith {
        removeGoggles _target;
        _unit addItem _previousGoggles;
    };

    if ([_target, _previousGoggles] call CBA_fnc_canAddItem) exitWith {
        removeGoggles _target;
        _target addItem _previousGoggles;
    };

    // If the target's goggles can fit in neither unit's nor target's inventory, drop them on the ground
    private _weaponHolder = nearestObject [_target, "WeaponHolder"];

    if (isNull _weaponHolder || {_target distance _weaponHolder > 2}) then {
        _weaponHolder = createVehicle ["GroundWeaponHolder", [0, 0, 0], [], 0, "NONE"];
        _weaponHolder setPosASL getPosASL _target;
    };

    _weaponHolder addItemCargoGlobal [_previousGoggles, 1];
};

```

Reply:
My suggestion can probably be optimised, as some of the code is identical.